src/gui/transferlistwidget.cpp

```diff
@@ -208,9 +210,14 @@ inline QModelIndex TransferListWidget::mapFromSource(const QModelIndex &index) c
     return nameFilterModel->mapFromSource(index);
 }
 
-void TransferListWidget::torrentDoubleClicked(const QModelIndex& index)
+void TransferListWidget::torrentDoubleClicked()
 {
-    BitTorrent::TorrentHandle *const torrent = listModel->torrentHandle(mapToSource(index));
+    const QModelIndexList selectedIndexes = selectionModel()->selectedRows();
+    if (selectedIndexes.size() != 1) return;
+    if (!selectedIndexes.first().isValid()) return;
+
+    const QModelIndex index = listModel->index(mapToSource(selectedIndexes.first()).row());
```

Comment:
I would break it after the assignment:
```C++
BitTorrent::TorrentHandle *const torrent = 
    listModel->torrentHandle(listModel->index(mapToSource(selectedIndexes.first()).row()));
```